app/javascript/components/admin/exercises/Edit.jsx

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+import Alert from 'react-bootstrap/Alert'
+import { Link } from "react-router-dom";
+import { defaultHeaders } from "../../../utils/request";
+
+class Edit extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      name: "",
+      description: "",
+      errors: {
+        name: [],
+        description: []
+      },
+      success: false
+    };
+
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.stripHtmlEntities = this.stripHtmlEntities.bind(this);
```

Comment:
I don't believe this function is actually used anywhere, the tutorial added it but never used it.

Reply:
Have removed this function now